src/sagemaker_xgboost_container/algorithm_mode/serve.py

```diff
@@ -164,12 +165,11 @@ def _parse_accept(request):
     :param request: flask request
     :return: parsed accept type
     """
-    accept, _ = cgi.parse_header(request.headers.get("accept").lower())
-    if not any(accept in mimetypes for mimetypes in ["application/json", "application/jsonlines",
-                                                     "application/x-recordio-protobuf", "text/csv"]):
-        if sm_env_constants.SAGEMAKER_DEFAULT_INVOCATIONS_ACCEPT in os.environ:
-            return os.getenv('SAGEMAKER_DEFAULT_INVOCATIONS_ACCEPT')
-        raise ValueError("Accept type {} is not supported.".format(accept))
+    default_accept = os.getenv('SAGEMAKER_DEFAULT_INVOCATIONS_ACCEPT')
+    accept, _ = cgi.parse_header(request.headers.get("accept", default_accept))
```

Comment:
Does the result of `request.headers.get("accept", default_accept)` need to be lowercased still?

Reply:
Thanks Patrick, updated to lowercase and to improve the error handling.